packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts

```diff
@@ -31,47 +32,69 @@ export class HostedPluginDeployerHandler implements PluginDeployerHandler {
     /**
      * Managed plugin metadata backend entries.
      */
-    private readonly currentBackendPluginsMetadata: PluginMetadata[] = [];
+    private currentBackendPluginsMetadata: PluginMetadata[] = [];
 
     /**
      * Managed plugin metadata frontend entries.
      */
-    private readonly currentFrontendPluginsMetadata: PluginMetadata[] = [];
+    private currentFrontendPluginsMetadata: PluginMetadata[] = [];
 
-    getDeployedFrontendMetadata(): PluginMetadata[] {
+    private backendPluginsMetadataDeferred = new Deferred<void>();
+
+    private frontendPluginsMetadataDeferred = new Deferred<void>();
+
+    async getDeployedFrontendMetadata(): Promise<PluginMetadata[]> {
+        // await first deploy
+        await this.frontendPluginsMetadataDeferred.promise;
+        // fetch the last deployed state
         return this.currentFrontendPluginsMetadata;
     }
 
-    getDeployedBackendMetadata(): PluginMetadata[] {
+    async getDeployedBackendMetadata(): Promise<PluginMetadata[]> {
+        // await first deploy
+        await this.backendPluginsMetadataDeferred.promise;
+        // fetch the last deployed state
         return this.currentBackendPluginsMetadata;
     }
 
     async deployFrontendPlugins(frontendPlugins: PluginDeployerEntry[]): Promise<void> {
+        const pluginsMetadata: PluginMetadata[] = []; // reset plugins on each deploy
```

Comment:
sorry resetting was wrong, now old plugins get uninstalled on dynamically installing a new one, so introducing a deferred to await first deployment is enough

Reply:
Done